### ISSUE TITLE
Generate a more compact index

### ISSUE THREAD

**Issue by martinthomson** [ietf-tools/xml2rfc#862]
The current method for constructing an index uses `<xref>` elements underneath, which generates an extremely verbose index if you choose to use that feature.  Take [this example](https://ietf-wg-ohai.github.io/oblivious-http/index/draft-ietf-ohai-ohttp.html#name-index) (this link won't work forever, sorry, but [here is an archive](https://github.com/ietf-wg-ohai/oblivious-http/suites/7847121802/artifacts/332202039)), which shows just how bad this can get.

A more compact version of the same index might uses symbols in place of the words.

**Comment by jrlevine** [comment]
Assuming you don't plan to print it out, why is this a problem?

**Comment by cabo** [comment]
On 2022-08-17, at 16:40, John L ***@***.***> wrote:
> 
> Assuming you don't plan to print it out, why is this a problem?

Some of us actually look at indexes and click on things (did it just yesterday for the C++ standard).
Doing this in a forest of noise is unnecessarily hard.

Grüße, Carsten

**Comment by cabo** [comment]
On 2022-08-17, at 18:21, John L ***@***.***> wrote:
> 
> [Section 1, Paragraph 3]; [Section 2, Paragraph 1, 3, 6.1, 6.2, 6.3, 6.9, 6.10]

[§ 1: ¶ 3]; [§ 2: ¶ 1, 3, 6.1, 6.2, 6.3, 6.9, 6.10]

Now we are talking :-)

Grüße, Carsten

**Comment by martinthomson** [comment]
> [§ 1: ¶ 3]; [§ 2: ¶ 1, 3, 6.1, 6.2, 6.3, 6.9, 6.10]

I can give that a spin.  Note that we have three levels of nesting here, which will make it a little more challenging to present, but a single layer of collapse might help.

**Comment by martinthomson** [comment]
OK, so it turns out that Carsten's (or maybe it was John's) suggestion is still a good one, but not one that I'm likely to get around to.  It's a little tricky to manage.  Is there any reason that an optional feature like this would be undesirable?

**Comment by martinthomson** [comment]
I took a look at implementing the merging idea, which is a good one.  That is an entirely different part of the process.

My limited understanding is that the generation of links in the index occurs in the preptool.  Basically, in step 1 the preptool populates an index with a bunch of `<xref>` elements, then the index is rendered using fairly normal `<xref>` rendering rules maybe with a few tweaks.  This pull request only affects the rendering of those links into text or HTML.  A separate change would be needed to change the content of those links.  That might happen, but it can do so separately.

**Comment by reschke** [comment]
FWIW; the index should not be part of the prepped document.

**Comment by reschke** [comment]
>  the point of prepping is to get a document ready to render

Nope. That's how it's used by xml2rfc right now, but that has absolutely nothing to do what the preptool was designed for. Putting the Toc, the authors section and the index here is simply an abuse for how xml2rfc works internally.

**Comment by martinthomson** [comment]
I agree with @reschke.  The way that the preptool works currently leaks its internal business everywhere.
